### ISSUE TITLE
Could not pass event PlayerJoinEvent to TitleManager v2.1.5

### ISSUE THREAD

**Issue by mckite** [Puharesource/TitleManager#233]
**Describe the bug**
Console stacktrace
`Could not pass event PlayerJoinEvent to TitleManager v2.1.5`

**To Reproduce**
Seldom, When player joining

**Stacktrace / Error**
```
[11:25:39 ERROR]: Could not pass event PlayerJoinEvent to TitleManager v2.1.5
io.puharesource.mc.titlemanager.shaded.rx.exceptions.OnErrorNotImplementedException: Class io.puhare
source.mc.titlemanager.APIProvider$setHeaderAndFooter$1 can not access a member of class net.minecra
ft.server.v1_12_R1.PacketPlayOutPlayerListHeaderFooter with modifiers "private"
        at io.puharesource.mc.titlemanager.shaded.rx.internal.util.InternalObservableUtils$ErrorNotI
mplementedAction.call(InternalObservableUtils.java:386) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.internal.util.InternalObservableUtils$ErrorNotI
mplementedAction.call(InternalObservableUtils.java:383) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.internal.util.ActionSubscriber.onError(ActionSu
bscriber.java:44) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.observers.SafeSubscriber._onError(SafeSubscribe
r.java:153) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.observers.SafeSubscriber.onError(SafeSubscriber
.java:115) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.exceptions.Exceptions.throwOrReport(Exceptions.
java:212) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.observers.SafeSubscriber.onNext(SafeSubscriber.
java:139) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.internal.operators.OnSubscribeMap$MapSubscriber
.onNext(OnSubscribeMap.java:77) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.internal.operators.OnSubscribeFilter$FilterSubs
criber.onNext(OnSubscribeFilter.java:76) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.internal.operators.OnSubscribeFilter$FilterSubs
criber.onNext(OnSubscribeFilter.java:76) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.internal.operators.OperatorSubscribeOn$Subscrib
eOnSubscriber.onNext(OperatorSubscribeOn.java:74) ~[?:?]
        at io.puharesource.mc.titlemanager.event.BukkitEventObservingKt$observeEventRaw$1$executor$1
.execute(BukkitEventObserving.kt:39) ~[?:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78) ~[patched_1.12.2.
jar:git-Paper-1618]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12
.2.jar:git-Paper-1618]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:513) ~[patched_1
.12.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.PlayerList.onPlayerJoin(PlayerList.java:370) ~[patched_1.12
.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.PlayerList.a(PlayerList.java:183) ~[patched_1.12.2.jar:git-
Paper-1618]
        at protocolsupport.zplatform.impl.spigot.network.handler.SpigotLoginListenerPlay.joinGame(Sp
igotLoginListenerPlay.java:124) ~[?:?]
        at protocolsupport.protocol.packet.handler.AbstractLoginListenerPlay.tryJoin(AbstractLoginLi
stenerPlay.java:131) ~[?:?]
        at protocolsupport.protocol.packet.handler.AbstractLoginListenerPlay.tick(AbstractLoginListe
nerPlay.java:93) ~[?:?]
        at protocolsupport.zplatform.impl.spigot.network.handler.SpigotLoginListenerPlay.e(SpigotLog
inListenerPlay.java:70) ~[?:?]
        at net.minecraft.server.v1_12_R1.NetworkManager.a(NetworkManager.java:273) ~[patched_1.12.2.
jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.ServerConnection.c(ServerConnection.java:150) ~[patched_1.1
2.2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:948) ~[patched_1.12.
2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423) ~[patched_1.12.
2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:774) ~[patched_1.12.
2.jar:git-Paper-1618]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:666) ~[patched_1.1
2.2.jar:git-Paper-1618]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_221]
Caused by: java.lang.IllegalAccessException: Class io.puharesource.mc.titlemanager.APIProvider$setHe
aderAndFooter$1 can not access a member of class net.minecraft.server.v1_12_R1.PacketPlayOutPlayerLi
stHeaderFooter with modifiers "private"
        at sun.reflect.Reflection.ensureMemberAccess(Unknown Source) ~[?:1.8.0_221]
        at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(Unknown Source) ~[?:1.8.0_221]
        at java.lang.reflect.AccessibleObject.checkAccess(Unknown Source) ~[?:1.8.0_221]
        at java.lang.reflect.Field.set(Unknown Source) ~[?:1.8.0_221]
        at io.puharesource.mc.titlemanager.APIProvider$setHeaderAndFooter$1.invoke(APIProvider.kt:56
0) ~[?:?]
        at io.puharesource.mc.titlemanager.APIProvider$setHeaderAndFooter$1.invoke(APIProvider.kt:46
) ~[?:?]
        at io.puharesource.mc.titlemanager.extensions.ReflectionExtensionsKt.modify(ReflectionExtens
ions.kt:7) ~[?:?]
        at io.puharesource.mc.titlemanager.APIProvider.setHeaderAndFooter(APIProvider.kt:560) ~[?:?]

        at io.puharesource.mc.titlemanager.APIProvider.setHeaderWithPlaceholders(APIProvider.kt:526)
 ~[?:?]
        at io.puharesource.mc.titlemanager.TitleManagerPlugin.setHeaderWithPlaceholders(TitleManager
Plugin.kt) ~[?:?]
        at io.puharesource.mc.titlemanager.extensions.PlayerExtensionsKt.setPlayerListHeader(PlayerE
xtensions.kt:109) ~[?:?]
        at io.puharesource.mc.titlemanager.APIProvider$toHeaderAnimation$4.invoke(APIProvider.kt:340
) ~[?:?]
        at io.puharesource.mc.titlemanager.APIProvider$toHeaderAnimation$4.invoke(APIProvider.kt:46)
 ~[?:?]
        at io.puharesource.mc.titlemanager.animations.PartBasedSendableAnimation.update(PartBasedSen
dableAnimation.kt:81) ~[?:?]
        at io.puharesource.mc.titlemanager.animations.PartBasedSendableAnimation.start(PartBasedSend
ableAnimation.kt:56) ~[?:?]
        at io.puharesource.mc.titlemanager.TitleManagerPlugin.setHeaderFromText(TitleManagerPlugin.k
t:633) ~[?:?]
        at io.puharesource.mc.titlemanager.TitleManagerPlugin.access$setHeaderFromText(TitleManagerP
lugin.kt:44) ~[?:?]
        at io.puharesource.mc.titlemanager.TitleManagerPlugin$registerListeners$19.call(TitleManager
Plugin.kt:476) ~[?:?]
        at io.puharesource.mc.titlemanager.TitleManagerPlugin$registerListeners$19.call(TitleManager
Plugin.kt:44) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.internal.util.ActionSubscriber.onNext(ActionSub
scriber.java:39) ~[?:?]
        at io.puharesource.mc.titlemanager.shaded.rx.observers.SafeSubscriber.onNext(SafeSubscriber.
java:134) ~[?:?]
        ... 21 more
```

**Server information:**
 - TitleManager version  2.1.5
 - Spigot version: 1.12.2

**Comment by AlekseyTitov99** [comment]
same issue

**Comment by KCookie16** [comment]
same issue up!

**Comment by Puharesource** [comment]
Should be fixed with the latest version (2.1.6)

**Comment by Ant4rez** [comment]
Same error here: https://hastebin.com/gifabocato.bash

**Server information:**

    TitleManager version 2.1.6
    Spigot version: 1.8.8
